lib/TransportRPCClient.js

```diff
@@ -134,24 +130,31 @@ class TransportRPCClient extends Base.TransportRPCClient {
       _cb(err, body)
     }
 
-    const req = socket.post(opts, (err, res, body) => {
-      if (err) {
+    try {
+      const resp = await fetch(opts.url, {
+        ...opts,
+        method: 'POST'
+      })
+
+      const body = await resp.buffer()
+
+      if (!resp.ok) {
+        const err = new Error(body)
```

Comment:
i think `Error` constructor's parameter should be a string https://nodejs.org/api/errors.html#new-errormessage-options